common/membership/grpc_resolver.go

```diff
@@ -61,16 +59,12 @@ func initializeBuilder(monitor Monitor) GRPCResolver {
 	return GRPCResolver{}
 }
 
-func (g *GRPCResolver) MakeURL(service primitives.ServiceName) string {
```

Comment:
Please revert this part and make this a method on GRPCResolver again, and make the string private again. The comment on the type (line 39/41) explains why it was done this way: to force a dependency on GRPCResolver so that callers can't construct/use a "membership" url without something being stored in `globalGrpcBuilder.monitor`.

Leaving just the empty type in place but without actually requiring it to construct the URL defeats the purpose.